tooling/nargo/src/ops/test.rs

```diff
@@ -68,17 +69,27 @@ where
                     std::env::var("NARGO_IGNORE_TEST_FAILURES_FROM_FOREIGN_CALLS")
                         .is_ok_and(|var| &var == "true");
 
-                let mut foreign_call_log = ForeignCallLog::from_env("NARGO_TEST_FOREIGN_CALL_LOG");
+                let writer: Box<dyn std::io::Write> =
+                    match std::env::var("NARGO_TEST_FOREIGN_CALL_LOG") {
+                        Err(_) => Box::new(std::io::empty()),
+                        Ok(s) if s == "stdout" => Box::new(std::io::stdout()),
+                        Ok(s) => Box::new(
+                            OpenOptions::new()
+                                .create_new(true)
```

Comment:
I think this means `examples/oracle_transcript/log_and_exec_transcript.sh` should be modified to delete any existing file, otherwise it will fail to create it. 

The previous version was probably equivalent to `.create(true).truncate(true)`, so if there were multiple tests matched then the last one would win, but now the second test will fail, which is potentially safer in a way that it forces the user to run 1 test at a time.